JE-46109 add onBeforeInstallAddon/onAfterInstallAddon

vlobzakov opened the pull request:
JE-46109 add onBeforeInstallAddon/onAfterInstallAddon events. Requires to add missed descriptions of some parameters

vlobzakov committed:
add onBeforeInstallAddon/onAfterInstallAddon

JE-46109 add onBeforeInstallAddon/onAfterInstallAddon events. Requires to add missed descriptions of some parameters

vlobzakov committed:
JE-46109 onBeforeInstallAddon/onAfterInstallAddon

Removed extra dots and Jira's IDs.
Some parameters require definition. These are lines:
#1644,
#1667,
#1669,
#1671,
#1673,
#1678

vlobzakov committed:
Merge branch 'master' into vlobzakov-JE-46109-onBeforeInstallAddon-onAfterInstallAddon

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+Event Placeholders:  
+
+-   `${event.params.}`:  
+    - `settings` - ***What is it***  
```
values of http://docs.cloudscripting.com/creating-manifest/visual-settings/

vlobzakov replied:
- `settings` - custom form with predefined user input elements

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it***  
```
remove

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it***  
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it***  
```
remove

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it***  
+    - `envName` - short domain name (without hosting provider URL)  
+    - `actionKey` - ***What is it***  
```
remove

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+    - `envName` - short domain name (without hosting provider URL)  
+    - `actionKey` - ***What is it***  
+    - `id` - add-on id  
+    - `lang` - ***What is it***  
```
user's language code (https://en.wikipedia.org/wiki/Language_code) which was used during the installation. Optional parameter

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+The event is executed after add-on installation  
+  
+-   `${event.params.}`:  
+    - `settings` - ***What is it?***  
```
values of http://docs.cloudscripting.com/creating-manifest/visual-settings/

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it?***  
```
remove

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it?***  
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it?***  
```
remove

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it?***  
+    - `envName` - short domain name (without hosting provider URL)  
+    - `actionKey` - ***What is it?***  
```
remove

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
+    - `envName` - short domain name (without hosting provider URL)  
+    - `actionKey` - ***What is it?***  
+    - `id` - add-on id  
+    - `lang` - ***What is it?***  
```
user's language code (https://en.wikipedia.org/wiki/Language_code) which was used during the installation. Optional parameter

vlobzakov replied:
done.

vlobzakov committed:
Requested changes applied

Is line correct: https://github.com/vlobzakov/docs/edit/vlobzakov-JE-46109-onBeforeInstallAddon-onAfterInstallAddon/docs/creating-manifest/events.md#L1784 ?

vlobzakov reviewed (commented):
Requested changes done. Please review the line: https://github.com/vlobzakov/docs/edit/vlobzakov-JE-46109-onBeforeInstallAddon-onAfterInstallAddon/docs/creating-manifest/events.md#L1784 is it correct?

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,93 @@ onAfterRemoveSSL:
+    - `appid` - unique environment *appid* at the Jelastic Platform  
+    - `successText`- message that appears once action is successfully performed  
+  
+Subscription:  
```
change to 'Events Subscription Example'. Just 'Subscription' is confusing a little bit.

vlobzakov replied:
done

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,93 @@ onAfterRemoveSSL:
+type: update  
+name: Test Add-on Events Subscription  
+  
+onBeforeInstallAddon [nodeGroup:cp, id:test123]:  
```
replace *test123* with *yourAddonId* (all occurrences)

vlobzakov replied:
I replaced addon id 'test123' with 'myaddon'. 
And I propose to add a 'unique' word in front of add-on id for better understanding that add-ons' ids cannot coincide

vlobzakov committed:
addon's id replaced

I think it would be better to add a "unique" word for add-on id.

vlobzakov reviewed (commented):
please check the updates

SlavaKatiukha commented on docs/creating-manifest/events.md:
```diff
@@ -1757,7 +1757,93 @@ onAfterRemoveSSL:
+    - `id` - unique add-on id  
+    - `lang` - [user's language code](https://en.wikipedia.org/wiki/Language_code) which was used during the installation. Optional parameter  
+    - `region` - region, where an environment will be installed  
+    - `targetAppid`  - application identifier  
```
change to "unique environment *appid* at the Jelastic Platform"

vlobzakov replied:
done:
https://github.com/vlobzakov/docs/pull/19/commits/794d0b8d68d7c3a24eddcc5433116766cc222e61
https://github.com/vlobzakov/docs/pull/19/commits/1ffa9f07646e16b7a20380bc3f4a29c34db5067b

vlobzakov committed:
targetAppid description updated

JE-46109 targetAppid description updated

vlobzakov committed:
targetAppid description updated
